### PR TITLE
add github workflows

### PR DESCRIPTION
**Pull request by andyborn**
- add github lint/test workflows
 - use common makefile
 - bump deps

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -5,5 +5,5 @@ go 1.15
 require (
 	github.com/onsi/ginkgo v1.14.0
```
Comment (andyborn) [inline go.mod:6]: could remove these test deps maybe (in separate MR)?

File: .github/workflows/lint.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+name: Lint
```
Comment (andyborn) [inline .github/workflows/lint.yml:1]: these actions dont appear to be triggering yet, maybe they need to be set up on repo side too (but i wouldnt have permissions).

File: go.mod (hunk 1 above)
Comment (dim) [reply go.mod:6]: replace with `bsm` ones at least pls

File: .github/workflows/lint.yml (hunk 2 above)
Comment (dim) [reply .github/workflows/lint.yml:1]: instead of PR'ing within your own `andyborn:hk/modernise` fork, you should be (now) able to PR directly in this repo

**Comment by andyborn** [comment]
will change to branch from bsm repo

**Pull request closed without merging**